docs/agent-reference/agent-core/configuration-manager.md

```diff
@@ -0,0 +1,95 @@
+# Device Update Agent Configuration
+
+## APIs
+
+```c
+// A config object handle.
+typedef void* CONFIG_HANDLE
+
+// Instancitate a config object from specified file.
+CONFIG_HANDLE LoadConfig(const char * configFilePath);
+
+// Free all resources allocated for the config object.
+void FreeConfig(CONFIG_HANDLE config);
+
+// Returns value of the specified config name.
+// (Output is a json string).
+const char* GetValue(const char* name);
+
+```
+
+## Configuration File Format
```

Comment:
Are any of these values optional? If so, do we internally set the default value in the agent?

Reply:
Some are, will make that clear in the doc.